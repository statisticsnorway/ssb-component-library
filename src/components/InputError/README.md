InputError
========

> Text wrapper for error messages

### Usage

#### HTML

```html
<div class="ssb-input-error"><span>Insert error message</span></div>
<div class="ssb-input-error negative"><span>Insert error message</span></div>
```

#### React

```jsx harmony
<InputError
  errorMessage="Insert error message"
  negative={false}
/>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| errorMessage | string | Renders text |
| negative | bool | Changes text color |
