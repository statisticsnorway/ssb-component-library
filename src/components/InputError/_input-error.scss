.ssb-input-error {
  @include roboto;
  background: $ssb-red-3;
  color: $ssb-white;
  display: flex;
  font-size: 14px;
  font-weight: bold;
  justify-content: center;
  line-height: normal;
  margin-top: 12px;
  padding: 4px 8px;
  position: relative;

  &.negative {
    color: $ssb-dark-5;
  }

  &:before {
    border-color: transparent transparent $ssb-red-3 transparent;
    border-style: solid;
    border-width: 8px;
    content: '';
    position: absolute;
    top: -16px;
  }
}
