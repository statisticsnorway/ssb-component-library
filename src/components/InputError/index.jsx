import React from 'react';
import PropTypes from 'prop-types';

const InputError = ({ errorMessage, negative }) => (
	<span className={`ssb-input-error ${negative ? 'negative' : ''}`}>{errorMessage}</span>
);

InputError.defaultProps = {
	negative: false,
};

InputError.propTypes = {
	errorMessage: PropTypes.string.isRequired,
	negative: PropTypes.bool,
};

export default InputError;
