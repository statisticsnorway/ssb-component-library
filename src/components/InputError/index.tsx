import React from 'react'
import classNames from '../../utils/utils'

interface InputErrorProps {
  className?: string
  errorMessage: string
  negative?: boolean
  id?: string
}

const InputError: React.FC<InputErrorProps> = ({ className, errorMessage, negative = false, id }) => (
  <div className={classNames('ssb-input-error', negative && 'negative', className)} id={id}>
    <span aria-live='assertive'>{errorMessage}</span>
  </div>
)

export default InputError
