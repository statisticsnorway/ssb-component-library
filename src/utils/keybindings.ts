export const KEY_ARROW_LEFT = 37
export const KEY_ARROW_UP = 38
export const KEY_ARROW_RIGHT = 39
export const KEY_ARROW_DOWN = 40

export const KEY_ESCAPE = 27
export const KEY_ENTER = 13
export const KEY_BACKSPACE = 8
export const KEY_TAB = 9
export const KEY_SPACE = 32

export const KEY_SHIFT = 16
export const KEY_CTRL = 17
export const KEY_ALT = 18
export const KEY_META = 91
