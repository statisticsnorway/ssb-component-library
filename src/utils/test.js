import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
