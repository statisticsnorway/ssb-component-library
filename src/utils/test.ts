import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
