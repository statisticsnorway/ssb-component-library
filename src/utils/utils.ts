// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
