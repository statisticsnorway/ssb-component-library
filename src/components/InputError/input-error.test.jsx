import React from 'react'
import { render } from '../../utils/test'
import InputError from './index'

describe('InputError component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<InputError errorMessage='Insert error message' />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Toggles classNames correctly', () => {
    const { asFragment } = render(<InputError errorMessage='Insert error message' negative />)
    expect(asFragment()).toMatchSnapshot()
  })
})
