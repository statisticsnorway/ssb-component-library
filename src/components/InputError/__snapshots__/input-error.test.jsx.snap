// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ssb-input-error"
  >
    <span
      aria-live="assertive"
    >
      Insert error message
    </span>
  </div>
</DocumentFragment>
`;

exports[`InputError component Toggles classNames correctly 1`] = `
<DocumentFragment>
  <div
    class="ssb-input-error negative"
  >
    <span
      aria-live="assertive"
    >
      Insert error message
    </span>
  </div>
</DocumentFragment>
`;
